File: P1/ej4/filters/email_filters.py
from .base import Filter

class EmailFormatFilter(Filter):
    def execute(self, request):
        email = request.get('email', '')
        if '@' not in email:
            return "El correo debe contener el carácter @"
        username = email.split('@')[0]
        if not username:
            return "El correo debe contener texto antes del carácter @"
        return None

class EmailDomainFilter(Filter):
    def execute(self, request):
        email = request.get('email', '')
        if '@' not in email:
            return "El correo debe contener el carácter @"
        domain = email.split('@')[1].lower()
        allowed_domains = ['gmail.com', 'hotmail.com']
        if domain not in allowed_domains:
            return f"El dominio del correo debe ser uno de los siguientes: {', '.join(allowed_domains)}"
        return None

File: P2/ejercicio1/lib/filters/email_filters.dart
import 'base.dart';

class EmailFormatFilter extends Filter {
  @override
  String? execute(Map<String, String> request) {
    String email = request['email'] ?? '';
    if (email.indexOf('@') != email.lastIndexOf('@')) {
      return "El correo debe contener exactamente un carácter @";
    }
    String username = email.split('@')[0];
    if (username.isEmpty) {
      return "El correo debe contener texto antes del carácter @";
    }
    return null;
  }
}

class EmailDomainFilter extends Filter {
  @override
  String? execute(Map<String, String> request) {
    String email = request['email'] ?? '';
    String domain = email.split('@')[1].toLowerCase();

    final domainRegex = RegExp(r'^[a-z0-9.-]+\.[a-z]{2,10}$');

    if (domainRegex.hasMatch(domain)) {
      return "El dominio del correo debe tener un formato válido (ej: gmail.com, outlook.co.es, yahoo.es)";
    }
    return null;
  }
}
